.gitlab/dev_envs.yml

```diff
@@ -0,0 +1,91 @@
+.linux_dev_env_amd64:
+  tags: ["runner:docker"]
+  image: "$BUILDENV_REGISTRY/images/docker:20.10-py3"
+  variables:
+    BASE_IMAGE_TAG: v47431157-7a59834c
+    BASE_IMAGE_ARCH: x64
+    TARGET_IMAGE_ARCH: amd64
+
+.linux_dev_env_arm64:
+  tags: ["runner:docker-arm", "platform:arm64"]
+  image: "$BUILDENV_REGISTRY/images/docker:20.10-py3"
+  variables:
+    BASE_IMAGE_TAG: v47431157-7a59834c
+    BASE_IMAGE_ARCH: arm64
+    TARGET_IMAGE_ARCH: arm64
+
+.build_dev_env:
+  script:
+    # Docker Hub login
+    - DOCKER_REGISTRY_LOGIN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.docker_hub_login --with-decryption --query "Parameter.Value" --out text)
+    - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.docker_hub_pwd --with-decryption --query "Parameter.Value" --out text | docker login --username "$DOCKER_REGISTRY_LOGIN" --password-stdin docker.io
+    # Build
+    - docker build dev-envs/$PLATFORM --build-arg ARCH=$BASE_IMAGE_ARCH --build-arg TAG=$BASE_IMAGE_TAG
+
+.publish_dev_env:
+  script:
+    # Docker Hub login
+    - DOCKER_REGISTRY_LOGIN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.docker_hub_login --with-decryption --query "Parameter.Value" --out text)
+    - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.docker_hub_pwd --with-decryption --query "Parameter.Value" --out text | docker login --username "$DOCKER_REGISTRY_LOGIN" --password-stdin docker.io
+    # Build and push
+    - docker build dev-envs/$PLATFORM --build-arg ARCH=$BASE_IMAGE_ARCH --build-arg TAG=$BASE_IMAGE_TAG --tag registry.ddbuild.io/ci/datadog-agent-buildimages/dev-env-$PLATFORM:$TARGET_IMAGE_ARCH
```

Comment:
We can't use crane in this area because we have to push a multi-architecture image in the end. To your point however I was erroneously extending a block that was unnecessary so I removed that!